epigraph-cli: `db_connect` and top-level `use sqlx::PgPool;` not gated behind `db` feature

## Summary

`crates/epigraph-cli/src/lib.rs` declares the `db` feature (which gates `epigraph-db` + `sqlx` deps) but the lib itself uses `sqlx::PgPool` unconditionally. Consumers that try `default-features = false` to skip the heavy DB stack fail to compile.

## Reproduction

```toml
[dependencies]
epigraph-cli = { git = "https://github.com/epigraph-io/epigraph", branch = "main", default-features = false }
```

```
$ cargo build
error[E0432]: unresolved import `sqlx`
 --> .../epigraph-cli/src/lib.rs:3:5
  |
3 | use sqlx::PgPool;
  |     ^^^^ use of unresolved module or unlinked crate `sqlx`
```

## Source

`crates/epigraph-cli/src/lib.rs`:

```rust
pub mod enrichment;

use sqlx::PgPool;       // <-- unconditional
use std::sync::Arc;

pub async fn db_connect() -> Result<PgPool, Box<dyn std::error::Error>> {
    let url = std::env::var("DATABASE_URL").map_err(|_| {
        "DATABASE_URL not set — set it to postgresql://epigraph:epigraph@127.0.0.1:5432/epigraph"
    })?;
    Ok(PgPool::connect(&url).await?)
}
```

```toml
[features]
default = ["db"]
db = ["dep:epigraph-db", "dep:sqlx"]
```

## Suggested fix

Gate the import and the function behind `#[cfg(feature = "db")]`:

```rust
pub mod enrichment;

#[cfg(feature = "db")]
use sqlx::PgPool;
use std::sync::Arc;

#[cfg(feature = "db")]
pub async fn db_connect() -> Result<PgPool, Box<dyn std::error::Error>> {
    let url = std::env::var("DATABASE_URL").map_err(|_| {
        "DATABASE_URL not set — set it to postgresql://epigraph:epigraph@127.0.0.1:5432/epigraph"
    })?;
    Ok(PgPool::connect(&url).await?)
}
```

`embedding_service()` doesn't depend on sqlx and can stay unconditional.

## Why it matters

Surfaced while building `epigraph-claude-cli` (private overlay providing `ClaudeCliClient`) which only needs the `LlmClient` trait and `LlmError` from the `enrichment` module — none of the DB plumbing. Worked around by enabling default features (pulling in `sqlx`, `epigraph-db`, and the full DB dep tree just to satisfy the `use` statement). Any future consumer that wants the trait without the DB stack will hit the same wall.

Effort: 2 lines + recompile. Test: `cargo build -p epigraph-cli --no-default-features` should succeed after the fix.

--- a/crates/epigraph-cli/src/lib.rs
+++ b/crates/epigraph-cli/src/lib.rs
@@ -1,9 +1,11 @@
 pub mod enrichment;
 
+#[cfg(feature = "db")]
 use sqlx::PgPool;
 use std::sync::Arc;
 
 /// Connect to postgres via DATABASE_URL environment variable.
+#[cfg(feature = "db")]
 pub async fn db_connect() -> Result<PgPool, Box<dyn std::error::Error>> {
     let url = std::env::var("DATABASE_URL").map_err(|_| {
         "DATABASE_URL not set — set it to postgresql://epigraph:epigraph@127.0.0.1:5432/epigraph"

